rct229/rules/section6/section6rule7.py

```diff
@@ -0,0 +1,107 @@
+        super(Section6Rule7, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section6Rule7.ZoneRule(),
+            index_rmr="proposed",
+            id="6-7",
+            description="Proposed building is modeled with daylighting controls directly or through schedule adjustments.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*]",
+        )
+
+    class ZoneRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section6Rule7.ZoneRule, self,).__init__(
+                required_fields={"$": ["spaces", "surfaces"]},
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+            )
+
+        def get_calc_vals(self, context, data=None):
+            zone_p = context.proposed
+
+            daylight_flag_p = (
+                len(
+                    find_all(
+                        "$..surfaces[?adjacent_to = 'EXTERIOR'].subsurfaces[?classification != 'DOOR']",
+                        zone_p,
+                    )
+                )
+                > 0
+            )
+
+            has_daylight_control_flag = (
+                len(
+                    find_all(
+                        "$..spaces[*].interior_lighting[?daylighting_control_type!='NONE']",
+                        zone_p,
+                    )
+                )
+                > 0
+            )
+
+            daylight_schedule_adjustment_flag = (
+                len(
+                    find_all(
+                        "$..spaces[*].interior_lighting[?are_schedules_used_for_modeling_daylighting_control == true]",
+                        zone_p,
+                    )
+                )
+                > 0
+            )
+
+            return {
+                "daylight_flag_p": daylight_flag_p,
+                "has_daylight_control_flag": has_daylight_control_flag,
+                "daylight_schedule_adjustment_flag": daylight_schedule_adjustment_flag,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            daylight_flag_p = calc_vals["daylight_flag_p"]
+            has_daylight_control_flag = calc_vals["has_daylight_control_flag"]
+
+            return daylight_flag_p and has_daylight_control_flag
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            daylight_flag_p = calc_vals["daylight_flag_p"]
+            has_daylight_control_flag = calc_vals["has_daylight_control_flag"]
+            daylight_schedule_adjustment_flag = calc_vals[
+                "daylight_schedule_adjustment_flag"
+            ]
+
+            if daylight_flag_p and has_daylight_control_flag:
+                if daylight_schedule_adjustment_flag:
+                    manual_check_msg = MSG_WARN_DAYLIGHT
+                else:
+                    manual_check_msg = MSG_WARN_DAYLIGHT_NO_SCHEDULE
+
+            return manual_check_msg
```

Comment:
We tend to add the final logic to the return statement.

You do not need to worry about `if daylight_flag_p and has_daylight_control_flag:` because the `get_manual_check_required_msg` will only be called when `manual_check_required` is True.
This means, the `daylight_flag_p = True` and `has_daylight_control_flag = True` is the precondition for the `get_manual_check_required_msg` function in this case.

So you can rewrite this function to:
```python
def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
    daylight_schedule_adjustment_flag = calc_vals[
                "daylight_schedule_adjustment_flag"
            ]
    return MSG_WARN_DAYLIGHT if  daylight_schedule_adjustment_flag else MSG_WARN_DAYLIGHT_NO_SCHEDULE
```

Reply:
Great point! I learned new thing :).